### PR TITLE
Fix overflow when performing drop check calculations in NLL

### LINKED ISSUE #47589: [NLL] issue-29844.rs test overflows while adding drop-check rules
```
$ RUST_BACKTRACE=full rustc +stage1 src/test/run-pass/issue-29844.rs -Znll -Zborrowck=mir -Ztwo-phase-borrows
error[E0320]: overflow while adding drop-check rules for std::vec::Vec<AttachInfo<'_, Resources>>
  |
  = note: overflowed on DescriptorSet<'_>

error[E0320]: overflow while adding drop-check rules for std::vec::Vec<AttachInfo<'_, Resources>>
  |
  = note: overflowed on *const alloc::arc::ArcInner<DescriptorSet<'_>>

error[E0320]: overflow while adding drop-check rules for DescriptorSet<'_>
  |
  = note: overflowed on DescriptorSet<'_>

error[E0320]: overflow while adding drop-check rules for DescriptorSet<'_>
  |
  = note: overflowed on *const alloc::arc::ArcInner<DescriptorSet<'_>>

error: aborting due to 4 previous errors
```

### PR DESCRIPTION
**Pull request by Aaron1011**
Clearing out the infcx's region constraints after processing each type
ends up interacting badly with normalizing associated types. This commit
keeps all region constraints intact until the end of
TypeLivenessGenerator.add_drop_live_constraint, ensuring that normalized
types are able to re-use existing inference variables.

Fixes #47589

### CODE REVIEW

**Review by nikomatsakis — APPROVED** [review]

File: src/librustc_mir/borrow_check/nll/type_check/liveness.rs [hunk 1]
```diff
@@ -15,7 +15,11 @@ use dataflow::move_paths::{HasMoveData, MoveData};
 use rustc::mir::{BasicBlock, Location, Mir};
 use rustc::mir::Local;
 use rustc::ty::{self, Ty, TyCtxt, TypeFoldable};
+//use rustc::ty::subst::Kind;
```
Comment (Mark-Simulacrum) [inline src/librustc_mir/borrow_check/nll/type_check/liveness.rs]: nit: should be removed

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

**Comment by Aaron1011** [comment]
@Mark-Simulacrum: Fixed

**Pull request merged**